x-pack/plugins/security_solution/server/lib/siem_migrations/rules/data_stream/rule_migrations_data_stream.test.ts

```diff
@@ -0,0 +1,118 @@
+ */
+
+import { RuleMigrationsDataStream } from './rule_migrations_data_stream';
+import { Subject } from 'rxjs';
+import type { InstallParams } from '@kbn/data-stream-adapter';
+import { DataStreamSpacesAdapter } from '@kbn/data-stream-adapter';
+import { elasticsearchServiceMock } from '@kbn/core-elasticsearch-server-mocks';
+import { loggerMock } from '@kbn/logging-mocks';
+
+jest.mock('@kbn/data-stream-adapter');
+
+const MockedDataStreamSpacesAdapter = DataStreamSpacesAdapter as unknown as jest.MockedClass<
+  typeof DataStreamSpacesAdapter
+>;
+
+const esClient = elasticsearchServiceMock.createStart().client.asInternalUser;
+
+describe('SiemRuleMigrationsDataStream', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('constructor', () => {
+    it('should create DataStreamSpacesAdapter', () => {
+      new RuleMigrationsDataStream({ kibanaVersion: '8.13.0' });
+      expect(MockedDataStreamSpacesAdapter).toHaveBeenCalledTimes(1);
+    });
+
+    it('should create component templates', () => {
+      new RuleMigrationsDataStream({ kibanaVersion: '8.13.0' });
+      const [dataStreamSpacesAdapter] = MockedDataStreamSpacesAdapter.mock.instances;
+      expect(dataStreamSpacesAdapter.setComponentTemplate).toHaveBeenCalledWith(
+        expect.objectContaining({ name: '.kibana.siem-rule-migrations' })
+      );
+    });
+
+    it('should create index templates', () => {
+      new RuleMigrationsDataStream({ kibanaVersion: '8.13.0' });
+      const [dataStreamSpacesAdapter] = MockedDataStreamSpacesAdapter.mock.instances;
+      expect(dataStreamSpacesAdapter.setIndexTemplate).toHaveBeenCalledWith(
+        expect.objectContaining({ name: '.kibana.siem-rule-migrations' })
+      );
+    });
+  });
+
+  describe('install', () => {
+    it('should install data stream', async () => {
+      const dataStream = new RuleMigrationsDataStream({ kibanaVersion: '8.13.0' });
+      const params: InstallParams = {
+        esClient,
+        logger: loggerMock.create(),
+        pluginStop$: new Subject(),
+      };
+      await dataStream.install(params);
+      const [dataStreamSpacesAdapter] = MockedDataStreamSpacesAdapter.mock.instances;
+      expect(dataStreamSpacesAdapter.install).toHaveBeenCalledWith(params);
+    });
+
+    it('should log error', async () => {
+      const dataStream = new RuleMigrationsDataStream({ kibanaVersion: '8.13.0' });
+      const params: InstallParams = {
+        esClient,
+        logger: loggerMock.create(),
+        pluginStop$: new Subject(),
+      };
+      const [dataStreamSpacesAdapter] = MockedDataStreamSpacesAdapter.mock.instances;
+      const error = new Error('test-error');
+      (dataStreamSpacesAdapter.install as jest.Mock).mockRejectedValueOnce(error);
+
+      await dataStream.install(params);
+      expect(params.logger.error).toHaveBeenCalledWith(expect.any(String), error);
+    });
+  });
+
+  describe('installSpace', () => {
+    it('should install space', async () => {
```

Comment:
It means installing a datastream with space id, not installing the space itself, right?